crates/compiler/src/lib.rs

```diff
@@ -222,15 +221,17 @@ mod v2 {
             .try_into()
             .context("Converting to Sierra class")?;
 
-        validate_compatible_sierra_version(
-            &sierra_class,
-            ListSelector::ListName(
-                casm_compiler_v2::allowed_libfuncs::BUILTIN_ALL_LIBFUNCS_LIST.to_string(),
-            ),
-        )
-        .context("Validating Sierra class")?;
-
-        let casm_class = CasmContractClass::from_contract_class(sierra_class, true)
+        sierra_class
+            .validate_version_compatible(
+                cairo_lang_starknet_classes::allowed_libfuncs::ListSelector::ListName(
+                    cairo_lang_starknet_classes::allowed_libfuncs::BUILTIN_ALL_LIBFUNCS_LIST
+                        .to_string(),
+                ),
+            )
+            .context("Validating Sierra class")?;
+
+        // TODO: determine `max_bytecode_size`
```

Comment:
> Since only "starknet valid" classes should arrive here, we ought to be safe.

That's not entirely true though. DECLARE transactions submitted via `starknet_estimateFee` / `starknet_simulateTransactions` also contain a class that ends up being compiled here.